### DISCUSSION TITLE
Visual Studio 2022 17.4 Support

### DISCUSSION THREAD

**Discussion by jeromelaban** [unoplatform/uno#9513] [General]
### Opened issue(s)
- https://github.com/xamarin/xamarin-macios/issues/16524
- https://github.com/xamarin/xamarin-macios/issues/16400
- https://github.com/xamarin/xamarin-android/issues/7473

### Fixed
- ~~Preview 2.2 Mobile issues https://github.com/dotnet/maui/issues/10234~~
- ~~https://developercommunity.visualstudio.com/t/Unable-to-debug-MAUI-app-if-multiple-tar/10120600~~

**Comment by queequac** [comment]
https://github.com/xamarin/xamarin-android/issues/7473 is missing in the list, which prevents Android release builds with 17.4

**Reply by jeromelaban** [reply]
Indeed it was missing. A workaround is to add `-r android-arm64` for devices or `-r android-x86` / `-r android-x64` for simulators, on the `dotnet build` command line.

**Reply by queequac** [reply]
Unfortunately not a workaround for publishing apps and packaging as AAB. :)

**Reply by jeromelaban** [reply]
Of course, it's only to build locally until this gets fixed.

**Reply by queequac** [reply]
@jeromelaban  Just out of curiosity... what exactly is the generated Java code good for? It seems to be one type declaration for each Xaml Ui element. I am wondering why this needs some Java flavors for each and every UiElement class.🤔

**Reply by jeromelaban** [reply]
For our current implementation, java counterparts need to be generated because all types `UIElement` types are inheriting from `Android.Views.ViewGroup` as well.

**Reply by queequac** [reply]
Maybe let me rephrase the question: Why does this require all UiElement classes to bring their own Java bits along? The actual java code seems to be quite minimalistic, i.e. registration at the mono runtime and some reference counting. It is always the same code, just different type names passed as string.

I am just wondering why this needs to be done on a per-class basis in Java and not only once for UiElement? Furthermore I am not sure if the same could not be achieved by the means of interop mono provides through C#.

**Reply by jeromelaban** [reply]
This is how the .NET Android bindings work to have proper handling of java method overrides and similar features.

**Comment by HakanL** [comment]
I've struggled to get the ToDo sample application to work under 17.4 with .NET 7 as well. Issue raised here: https://github.com/unoplatform/uno.todo/issues/443
